sim/warlock/items.go

```diff
@@ -225,17 +213,31 @@ var ItemSetGladiatorsFelshroud = core.NewItemSet(core.ItemSet{
 // T13
 var ItemSetVestmentsOfTheFacelessShroud = core.NewItemSet(core.ItemSet{
 	Name: "Vestments of the Faceless Shroud",
-	Bonuses: map[int32]core.ApplyEffect{
-		2: func(agent core.Agent) {
+	Bonuses: map[int32]core.ApplySetBonus{
+		2: func(agent core.Agent, setBonusAura *core.Aura) {
 			warlock := agent.(WarlockAgent).GetWarlock()
 
-			warlock.AddStaticMod(core.SpellModConfig{
+			setBonusAura.AttachSpellMod(core.SpellModConfig{
 				Kind:      core.SpellMod_Cooldown_Flat,
 				TimeValue: -time.Minute * 4,
 				ClassMask: WarlockSpellSummonDoomguard | WarlockSpellSummonInfernal,
 			})
+
+			summonDuration := core.TernaryDuration(warlock.Spec == proto.Spec_SpecDemonologyWarlock, 20*time.Second, 30*time.Second)
+
+			setBonusAura.AttachSpellMod(core.SpellModConfig{
+				Kind:      core.SpellMod_BuffDuration_Flat,
+				ClassMask: WarlockSpellSummonDoomguard,
+				TimeValue: summonDuration,
+			})
+
+			setBonusAura.AttachSpellMod(core.SpellModConfig{
+				Kind:      core.SpellMod_BuffDuration_Flat,
+				ClassMask: WarlockSpellSummonInfernal,
+				TimeValue: summonDuration,
+			})
```

Comment:
I think this could be a single mod with `ClassMask: WarlockSpellSummonDoomguard | WarlockSpellSummonInfernal` ?

Reply:
Yep!